activesupport/lib/active_support/number_helper/number_to_rounded_converter.rb

```diff
@@ -16,16 +15,74 @@ def convert
           else
             @number = number.to_d
           end
+        end
+      end
 
-          if options.delete(:significant) && precision > 0
-            digits, rounded_number = digits_and_rounded_number(precision)
-            precision -= digits
-            precision = 0 if precision < 0 # don't let it be negative
+      def round
+        if precision
+          if options[:significant] && precision > 0
+            rounded_number = round_significant
           else
             rounded_number = number.round(precision)
             rounded_number = rounded_number.to_i if precision == 0 && rounded_number.finite?
             rounded_number = rounded_number.abs if rounded_number.zero? # prevent showing negative zeros
           end
+        else
+          rounded_number = number
+        end
+
+        rounded_number
+      end
```

Comment:
@bquorning I'm wondering if the method `round` (or maybe `rounded`) should just be an accessor to an instance variable set up in the initializer, where the rounding happens, so there is just one big if statement there.